sdk/src/Services/S3/Custom/_async/AmazonS3Client.Extensions.cs

```diff
@@ -147,7 +147,21 @@ Task ICoreAmazonS3.MakeObjectPublicAsync(string bucket, string objectKey, bool e
 
         Task ICoreAmazonS3.EnsureBucketExistsAsync(string bucketName)
         {
-            return this.PutBucketAsync(bucketName);
+            Task task = this.PutBucketAsync(bucketName);
```

Comment:
You should just use the async/await something like.

```
async Task ICoreAmazonS3.EnsureBucketExistsAsync(string bucketName)
{
   try
   {
        await this.PutBucketAsync(bucketName);
   }
   catch(BucketAlreadyOwnedByYouException)
   {
   }
}

Reply:
The exception in `this.PutBucketAsync(bucketName)` is thrown in another thread and based on my testing, catch block wouldn't handle it. Hence, I needed to resort to exception handling based on Microsoft's documentation. The only difference I see in your proposed code is addition of `async` to method signature. Let me give it a try.

Reply:
Fixed. It works. Will kick another dry-run.